### PR TITLE
optimize paged attention on triton3

### PR DESCRIPTION
**Pull request by grimoire**
triton3 has move the cuda fast math location.
This PR support fast expf in paged attention with triton3.0. 

> [!NOTE]  
> None-cuda backend end might not work.

The fill kv kernel and attention is updated so we can change kv layout in the future.

### CODE REVIEW

**Comment by grimoire** [comment]
```
--------------------------------------------------
total time: 683.11s
concurrency: 1, test_round: 1
input_tokens: 1000000, output_tokens: 1
first_token latency(min, max, ave): 679.787s, 679.787s, 679.787s
total_token latency(min, max, ave): 679.787s, 679.787s, 679.787s
token_latency percentiles(50%,75%,95%,99%)(s): [679.787, 679.787, 679.787, 679.787]
throughput(output): 0.0 token/s
throughput(total): 1463.9 token/s
--------------------------------------------------
```

File: lmdeploy/pytorch/kernels/cuda/pagedattention.py [hunk 1]
```diff
@@ -13,18 +13,15 @@
 logger = get_logger('lmdeploy')
 
 TRITON_VERSION = version.parse(triton.__version__)
+VERSION_300 = version.parse('3.0.0')
```
Comment (RunningLeon) [inline lmdeploy/pytorch/kernels/cuda/pagedattention.py:16]: may fix ut and conflicts

**Review by AllentDan — COMMENTED** [review]
May update UT for the new layout type.

File: lmdeploy/pytorch/kernels/cuda/w8a8_triton_kernels.py [hunk 2]
```diff
@@ -305,7 +312,7 @@ def _per_token_quant_int8(
     # Quant
     _absmax = tl.maximum(tl.max(tl.abs(y)), eps)
     y_s = _absmax / 127
-    y_q = tl.math.round(y / y_s).to(tl.int8)
+    y_q = tl_round(y / y_s).to(tl.int8)
```
[the remark this answers is not in the document]
Comment (grimoire) [reply lmdeploy/pytorch/kernels/cuda/w8a8_triton_kernels.py:315]: Is it true for negative y?

[the remark this answers is not in the document]
Comment (AllentDan) [reply lmdeploy/pytorch/kernels/cuda/w8a8_triton_kernels.py:315]: Oh, just found it is symmetric quantization.

**Review by RunningLeon — COMMENTED** [review]
flash_attn-2.6.3+cu118torch2.3cxx11abiFALSE-cp38-cp38-linux_x86_64.whl

**Review by AllentDan — APPROVED** [review]
LGTM

File: lmdeploy/pytorch/kernels/cuda/pagedattention.py [hunk 3]
```diff
@@ -38,7 +35,9 @@ def tanh(x):
     triton.Config({}, num_stages=2, num_warps=8),
     triton.Config({}, num_stages=2, num_warps=4),
 ],
-                 key=['BLOCK_H', 'BLOCK_N', 'BLOCK_DMODEL', 'BLOCK_DV'])
+                 key=['BLOCK_H', 'BLOCK_N', 'BLOCK_DMODEL', 'BLOCK_DV'],
+                 warmup=10,
+                 rep=25)
```
Comment (lvhan028) [inline lmdeploy/pytorch/kernels/cuda/pagedattention.py:40]: what does `rep` mean?

Comment (grimoire) [reply lmdeploy/pytorch/kernels/cuda/pagedattention.py:40]: repeat benchmark for autotuning.

Comment (lvhan028) [reply lmdeploy/pytorch/kernels/cuda/pagedattention.py:40]: How long does this setting `warmup=10, rep=25`  take?

Comment (grimoire) [reply lmdeploy/pytorch/kernels/cuda/pagedattention.py:40]: each tuning config take 34~35ms

**Review by RunningLeon — APPROVED** [review]
LGTM

**Pull request merged**